### ISSUE TITLE
Implement user specified visit slicing to allow multiple clients consuming the visit result 

### ISSUE THREAD

**Issue by jobergum** [vespa-engine/vespa#5055]
It would be nice if user can specify number of slices (e.g 25) when performing a large visitation operation in order to use parallel clients where each client will visit a subset of the total corpus size. The first call specifies the number of slices/partitions/fragments. e.g a request to partition the corpus into 25 roughly equal slices:

 GET /document/v1/namespace/doc-type/?slices=25&selection=ss   

Subsequent parallel calls (by continuation id) must then specify the slice id 

GET /document/v1/namespace/doc-type/?continuation=xxx&slice-id=0
GET /document/v1/namespace/doc-type/?continuation=xxx&slice-id=1
GET /document/v1/namespace/doc-type/?continuation=xxx&slice-id=24 

This would allow for higher throughput by allowing multiple clients to consume the streams.

**Comment by vekterli** [comment]
Since visitor sessions explicitly iterate over the linear super bucket space in a deterministic fashion, the most straight forward way of doing this is likely to skip super buckets that are not part of the session's assigned slice. With pseudo-random document distribution all super buckets are expected to be effectively equal in size, so this distributes session data to read "perfectly" (in theory).

Due to the way continuation tokens work internally today (effectively a set of cursors into the bucket space with some meta data related to high level bucket distribution), it's easier if we require specifying the slice count and slice id _per request_ instead of having an up-front request. I.e. a request would look like this:
```http
GET /document/v1/namespace/doc-type/?continuation=xxx&slices=25&slice-id=0
```
This also incidentally means clients can run entirely in parallel without any prior synchronization.

Assuming there are no concurrent distribution bit changes (only triggered by certain topology changes and may be hardcoded through config), sliced visit sessions shall be guaranteed to cover disjoint ranges of the data space.

**Comment by kkraune** [comment]
can't we just support something in the selection language that selects a part of the bucket space, like superbucketID MOD X ?  This is similar to what I do to dump 10% where I naively do docID MOD 10 - this scans all buckets and is inefficient, though

**Comment by vekterli** [comment]
That is definitely possible, but it has some caveats:
- Parallelism inherently becomes part of the selection (the domain specific query) rather than being orthogonal to it. That means you cannot parallelize any arbitrary full corpus selection expression without altering the structure of the expression itself.
- It further exposes, and thereby effectively makes an API requirement of, how we use buckets as the distribution mechanism for documents. This makes it harder if we want to transition to other approaches (tablets with a natural key ordering etc) in the future.
- It removes any abstractions of _how_ multi-client slicing is implemented, which ties into the previous point.

**Comment by jonmv** [comment]
https://blog.vespa.ai/sliced-visiting/
